From f36ef86e67185b16b65445092c40a3d75d0be45c Mon Sep 17 00:00:00 2001
From: Agriya Khetarpal <74401230+agriyakhetarpal@users.noreply.github.com>
Date: Fri, 8 Nov 2024 02:46:00 +0530
Subject: [PATCH] Modernise packaging

This patch introduces a PEP 517/518 compliant build system using setuptools and
declarative metadata in pyproject.toml, and fixes syntax errors in setup.py. The
sources for the extension module are correctly specified as Cython source files.

Some part of the credits go to the upstream
PR at https://github.com/LLNL/zfp/pull/234

Co-Authored-By: David M. Rogers <2243447+frobnitzem@users.noreply.github.com>
---
 pyproject.toml | 13 +++++++++++++
 setup.py       | 18 +++++++++---------
 2 files changed, 22 insertions(+), 9 deletions(-)
 create mode 100644 pyproject.toml

diff --git a/pyproject.toml b/pyproject.toml
new file mode 100644
index 0000000..88adcc8
--- /dev/null
+++ b/pyproject.toml
@@ -0,0 +1,13 @@
+[build-system]
+requires = ["setuptools", "cython", "numpy"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "zfpy"
+version = "1.0.1"
+authors = [
+    { name = "Peter Lindstrom", email = "zfp@llnl.gov" },
+    { name = "Danielle Asher", email = "zfp@llnl.gov" },
+]
+description = "zfp compression in Python"
+requires-python = ">=3.0"
diff --git a/setup.py b/setup.py
index fa2da6e..919fa7c 100644
--- a/setup.py
+++ b/setup.py
@@ -2,14 +2,14 @@ from setuptools import setup, Extension
 import numpy as np
 
 setup(
-    name="zfpy",
-    version="1.0.1",
-    author="Peter Lindstrom, Danielle Asher",
-    author_email="zfp@llnl.gov",
     url="https://zfp.llnl.gov",
-    description="zfp compression in Python",
-    long_description="zfp is a compressed format for representing multidimensional floating-point and integer arrays. zfp provides compressed-array classes that support high throughput read and write random access to individual array elements. zfp also supports serial and parallel compression of whole arrays using both lossless and lossy compression with error tolerances. zfp is primarily written in C and C++ but also includes Python and Fortran bindings.",
-    ext_modules=[Extension("zfpy", ["build/python/zfpy.c"],
-                           include_dirs=["include", np.get_include()],
-                           libraries=["zfp"], library_dirs=["build/lib64", "build/lib/Release"]), language_level = "3"]
+    ext_modules=
+        [
+            Extension(
+                name="zfpy",
+                sources=["python/zfpy.pyx"],
+                include_dirs=["include", np.get_include()],
+                libraries=["zfp"],
+                library_dirs=["build/lib64", "build/lib"])
+        ],
 )
-- 
2.39.3 (Apple Git-146)

